.github/workflows/pylint_pytest.yml

```diff
@@ -0,0 +1,57 @@
+name: pylint-and-pytest-checker
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+
+  testing:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: checkout repo
+        uses: actions/checkout@v4
+
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pytest
+          pip install -r requirements.txt
+
+      - name: Run all tests 
+        run: pytest .
+
+  linting:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: checkout repo
+        uses: actions/checkout@v4
+
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pylint
+        
+      - name: Run the linter on pipeline files
+        run: pylint --fail-under=8 ./pipeline
```

Comment:
Can't you just pylint *.py?

Reply:
just updated it in [a8a928f](https://github.com/Zhi-704/C11-Kappa-Group-Project/pull/35/commits/a8a928f48a2e077b0ea57678bb28b17fdf31b296), give this a look. *.py doesn't work as expected